### PR TITLE
Removed destroy() method from WaitCondition

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Partially fixes https://github.com/cogneco/ooc-kean/issues/459
I want to keep all the pull requests for https://github.com/cogneco/ooc-kean/issues/459 small, so here goes first part.

### CODE REVIEW

File: source/sdk/threading/native/ConditionUnix.ooc [hunk 1]
```diff
@@ -24,14 +24,13 @@ ConditionUnix: class extends WaitCondition {
     	raise("Something went wrong when calling pthread_cond_init")
   }
   free: override func {
-    this destroy()
+    pthread_cond_destroy(this _backend)
```
Comment (davidhesselbom) [inline source/sdk/threading/native/ConditionUnix.ooc]: Oooh, is this going to fix the unfixable memory leak from `_dl_allocate_tls`?

**Comment by davidhesselbom** [comment]
> I want to keep all the pull requests [...] small

Much appreciated.

File: source/sdk/threading/native/ConditionUnix.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/sdk/threading/native/ConditionUnix.ooc]: Sorry, I have no idea :(

**Comment by davidhesselbom** [comment]
There's probably nothing wrong with this, but I need to test it.

**Pull request merged**